lib/spack/spack/environment/environment.py

```diff
@@ -1345,12 +1235,18 @@ def _env_modifications_for_default_view(self, reverse=False):
 
         return all_mods, errors
 
-    def add_default_view_to_shell(self, shell):
-        env_mod = spack.util.environment.EnvironmentModifications()
+    def add_default_view_to_env(self, env_mod):
+        """
+        Collect the environment modifications to activate an environment using the
+        default view. Removes duplicate paths.
 
+        Args:
+            env_mod (spack.util.environment.EnvironmentModifications): the environment
+                modifications object that is modified.
+        """
         if default_view_name not in self.views:
             # No default view to add to shell
-            return env_mod.shell_modifications(shell)
+            return env_mod
```

Comment:
Minor, so can be done in a following PR. This method modifies the state of `env_mod` so it should not return it (gives the false impression that the return value is a different object than the argument). It's minor since peeking at code it seems we always discard the return value.

Reply:
it's a bit like a builder pattern

Reply:
It's not, I think. Here you pass the object as an argument and you return the same object. In the builder patter you have a method to get the result and other to build parts, but none of them takes as argument the object to be modified and returns it modified. 

With this:
```python
env1 = ...
env2 = ev.add_default_view_to_env(env1)
```
`env1` and `env2` are the same object. You modify one and you see the other modified.

Reply:
i agree, it makes more sense not to hint it is side effect free